SCIM documentation (#536)
configcat/docs: website/docs/advanced/team-management/scim/overview.mdx

---
id: scim-overview
title: (Beta) User provisioning (SCIM) Overview
description: This section is a step-by-step guide on how to configure User provisioning (SCIM) for your Organization in ConfigCat.
---

:::info
**Beta Feature**: SCIM provisioning is in public beta. It has been thoroughly tested with various Identity Providers. 
We're now collecting feedback from real-world usage to fine-tune the experience. Share your feedback [here](https://configcat.com/support).
:::

This section describes how to configure User provisioning (SCIM) for your Organization in ConfigCat.

User provisioning (SCIM) allows you to synchronize users and groups from your Identity Provider into a ConfigCat Organization. 
ConfigCat permissions then can be assigned to each synchronized Identity Provider (IdP) group, 
which puts their members to the selected ConfigCat permission groups upon a user provisioning process.

> Go to the *User provisioning (SCIM)* section on the <a href="https://app.configcat.com/organization/authentication/" target="_blank" rel="noopener noreferrer">Authentication & Provisioning</a> page to set up User provisioning. You need to be an Organization Admin to do this.

Author: sajnos nem könnyen

| ---------------------- | ------------------------- | ------------------------------------------------------- | ----------- | 
| John                   | Owners, Readers           | Organziation Admin, Billing Manager                     | As John is assigned to the Owners Identity Provider group that has the Organization Admin role, the Readers Identity Provider group's product level permissions won't be assigned as he is already an Organization Admin. |
| Jane                   | Readers, Billing Managers | Billing Manager, Product A/Readers, Product B/Readers   | A ConfigCat user can have the Billing Manager role and Product level permissions too, so the permissions are merged based on the Identity Provider groups. |
| Peter                  | Readers, Developers       | Product A/Developers, Product B/Developers              | As both the Readers and Developers Identity Provider groups have associated permissions to the same Products, the permissions will be merged based on the Identity Provider group's priority |

## 3. Start and stop provisioning for Identity Provider users  {#users}

> After a successful synchronization, the Identity Provider users will appear on the <a href="https://app.configcat.com/organization/authentication/" target="_blank" rel="noopener noreferrer">Authentication & Provisioning</a> page in the *3. Start and stop provisioning for Identity Provider users* section.

Author: igen

### What is managed by the provisioning?

The User Provisioning feature manages the following user related attributes after you start provisioning the Identity Provider user:
- **ConfigCat permissions**: with a fine-tuned provisioning configuration, you can easily manage your team members' ConfigCat permissions with your Identity Provider.
- **ConfigCat account's email address**: if an Identity Provider user's email address has been changed in the Identity Provider, the associated ConfigCat user's email address will be changed too and the ConfigCat user will only be able to log in with the new email address.
- **ConfigCat account's full name**: the Identity Provider user's full name is synchronized from the Identity Provider and overwrites the ConfigCat user's full name property.
- **Soft delete**: if an Identity Provider user is soft deleted in the Identity Provider, the ConfigCat account will be disabled and cannot be used.
- **Hard delete**: if an Identity Provider user is hard deleted in the Identity Provider, the ConfigCat account will be removed from the Organization entirely, but the ConfigCat account won't be deleted for a while.

### Start/stop provisioning

> In the Identity Provider users table, you can see a start/stop button in the last column. This indicates if an Identity Provider user's provisioning is active or not. 

Author: icon button

:::

If you stop the provisioning an Identity Provider user, their permissions will stay in the last state, and modifying provisioning permissions or changing the Identity Provider user's Identity Provider groups won't change their permissions. You'll be able to manually modify permissions for provisioning stopped Identity Provider users on the Team members/Members & Roles pages.  
After turning back the provisioning the permissions may change regarding the actual provisioning configurations.

### Provisioning status

The Provisioning status column in the Identity Provider users table shows you important information about the provisioning state.

#### Identity Provider user with an unverified email domain

The User Provisoning is synchronizing all Identity Provider users from your Identity Provider regardless if the Identity Provider user's email domain is a verified email domain in ConfigCat or not.  

> The Provisioning status will show an error message if the Identity Provider user's email address is not part of a verified domain in ConfigCat and you won't be able to start provisioning for Identity Provider user until you verify the email domain.

Reviewer: Itt szerintem rá kéne linkelni a Domain verifyoló oldalra vagy a doksijára.

- **Provisioning is in progress**: the provisioning is happening in a background job, so for a short period of time you can see the provisioning is in progress status. This can happen if your Identity Provider sends ConfigCat updates about your Identity Provider user or you modify your Identity Provider group's provisioning permissions.

## 4. Provision future users automatically {#auto-provisioning}

By default, after the Identity Provider synchronizes a new Identity Provider user to ConfigCat, the Identity Provider user will be in a provisioning stopped state, so you can verify your configuration/permissions and adjust your settings if needed.  
After you are confident that the provisioning works like you expect, you can turn the **Provision future users automatically** toggle to ON to immediately start the provisioning for newly synchronized Identity Provider users. 

<img className="bordered-img" src="/docs/assets/scim/dashboard/auto_provisioning.png" alt="Auto provisioning for new users" decoding="async" loading="lazy"/>

### Disable Domain Auto-assign

> After verifying your domain name in ConfigCat, all users signing up with that domain will be added to your Organization. With the auto-assign feature, you can also specify initial permissions for these users.  

Reviewer: ```suggestion
```

> User provisioning may interfere with ConfigCat's auto-assign feature. If you want to allow only provisioned users into your ConfigCat Organization, you should deactivate the auto-assign feature on Domain verification & Auto-assign users panel.

Reviewer: ```suggestion
ConfigCat has a few other features that might interfere with *User provisioning*:
- **Domain verification**: after verifying your domain name in ConfigCat, all users signing up with that domain will be added to your Organization.
- **Auto-assign**: the *auto-assign feature* assigns initial permissions to those users automatically.

If you want to allow only provisioned users into your ConfigCat Organization, you should deactivate the auto-assign feature in **Domain verification & Auto-assign users** panel.
```

<img className="bordered-img" src="/docs/assets/scim/dashboard/disable_auto_assign.png" alt="Disable domain auto-assign" decoding="async" loading="lazy"/>

## 5. Review ConfigCat users who are not managed by user provisioning  {#not-managed}

After configuring the user provisioning and synchronizing your Identity Provider users from your Identity Provider, it may happen that there are some ConfigCat accounts in your Organization who are not synchronized by the User Provisioning. 
There can always be exceptions, but in most cases:
- Either they no longer need access to ConfigCat - you may want to delete them.
- Or you want to manage them through your Identity Provider - in this case, go to your Identity Provider and set up sync for them.

> - Or you want to keep them this way - in this case, you need to manage their ConfigCat permissions on the Members & Roles page.

Reviewer: ```suggestion
- Or you want to keep them this way - in this case, you need to manage their ConfigCat permissions on the **Members & Roles** page.
```

Reviewer: szerintem ennek is linknek kéne lennie
